BLSQ/iaso
# Polio 1185 add check on user country

The endpoint could potentially allow users to PATCH campaigns for countries they don't have access to

Related JIRA tickets : POLIO-1185

## Self proofreading checklist


## Changes

- Use `Campaign.objects.filter_for_user(user)` to filter out campaigns user don't have access to
- Add test

## How to test

- Create a user with country permissions ("Localisation tab" in the user create/edit modal)
- Go to the admin and make gice this user staff status (not super user) 
- Find a campaign for a country not in the permissions list
- Go to `swagger`, find `api/polio/rounds/updatelqasfields`, click try it out
- Fill in the fields. `obr_name`should be from the campaign you looked up earlier
- You should get a 404

## Print screen / video

https://github.com/BLSQ/iaso/assets/38907762/18d394f3-8109-4232-952a-2c892ce6acc1


## Notes

This endpoint is not accessible via the iaso's UI: it's used by OpenHexa pipelines
